File: 792-binary-search/binary-search.java
class Solution {
    public int search(int[] nums, int target) {
        return helper(0, nums.length-1, nums, target);
    }
    public int helper(int low, int high, int[] nums, int target){
        if(low > high || high >=nums.length) return -1;
        int mid = low + (high - low)/2;
        
        if(target == nums[mid]) return mid;

        if(target < nums[mid]){
            return helper(low, mid-1, nums, target);
        }
        else{
            return helper(mid+1, high, nums, target);
        }
        
    }
}

File: 792-binary-search/binary-search.py
class Solution:
    def search(self, nums: List[int], target: int) -> int:
        return self.helper(0, len(nums)-1, nums, target)

    def helper(self, low, high, nums, target):
        mid= low + (high - low)//2

        if(high > len(nums) or high < low):
            return -1
        if(nums[mid] == target):
            return mid
        elif(nums[mid] < target):
            return self.helper(mid+1, high, nums, target)
        else:
            return self.helper(low, mid-1, nums, target)


    
        